clippy_lints/src/strings.rs

```diff
@@ -152,7 +161,21 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for StringLitAsBytes {
             if path.ident.name == "as_bytes" {
                 if let ExprKind::Lit(ref lit) = args[0].node {
                     if let LitKind::Str(ref lit_content, _) = lit.node {
-                        if lit_content.as_str().chars().all(|c| c.is_ascii()) && !in_macro(args[0].span) {
+                        let callsite = snippet(cx, args[0].span.source_callsite(), "");
+                        let expanded = format!("\"{}\"", lit_content.as_str());
+                        if callsite.starts_with("include_str!") {
+                            span_lint_and_sugg(
+                                cx,
+                                STRING_LIT_AS_BYTES,
+                                e.span,
+                                "calling `as_bytes()` on `include_str!(..)`",
+                                "consider using `include_bytes!(..)` instead",
+                                snippet(cx, args[0].span, r#""foo""#).replacen("include_str", "include_bytes", 1),
```

Comment:
Also not sure if this is the appropriate way to handle the replacement, I dont know how to destructure the callsite, Also, notably the output of line 164 and this are snippet function are the same, should I use arg[0].span rather than `callsite`?

I used replacen because I was afraid there may be other instances of "include_str" inside of the content of the span.

Reply:
I would replace the snippet function with `callsite`, to avoid building the snippet twice. But the default value of the `callsite` should be adapted to not just be an empty string. (either keep `r#""foo""#` or at least `r#""""#`)

The `replacen` seems good to me.